bedrock/base/geo.py

```diff
@@ -0,0 +1,72 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from django.conf import settings
+
+try:
+    import maxminddb
+except ImportError:
+    maxminddb = None
+
+
+GEO = None
+
+
+def _get_geo_client():
+    global GEO
+    if GEO is None and maxminddb is not None:
+        try:
+            GEO = maxminddb.open_database(settings.MAXMIND_DB_PATH)
+        except (maxminddb.InvalidDatabaseError, OSError, IOError):
+            pass
+
+    return GEO
+
+
+def get_country_from_ip(ip_addr):
+    """Return country info for the given IP Address."""
+    geo = _get_geo_client()
+    if geo is None and settings.DEV:
+        return settings.MAXMIND_DEFAULT_COUNTRY.upper()
+
+    if geo is not None:
+        try:
+            data = geo.get(ip_addr)
+        except ValueError:
+            data = None
+
+        if data:
+            country = data.get('country', data.get('registered_country'))
+            if country:
+                return country['iso_code'].upper()
+
+    return None
+
+
+def get_country_from_request_header(request):
+    """Return an uppercase 2 letter country code retrieved from request headers."""
+    if settings.DEV:
+        country_code = settings.DEV_GEO_COUNTRY_CODE
+    else:
+        country_code = request.META.get('HTTP_CF_IPCOUNTRY', 'XX')
+
+    if country_code == 'XX' or len(country_code) != 2:
+        return None
+
+    return country_code.upper()
+
+
+def get_country_from_maxmind(request):
+    ip_addr = request.META.get('HTTP_X_CLUSTER_CLIENT_IP',
```

Comment:
I honestly don't remember and we should check with @duallain about the proper header to check for this. This was part of the code I reverted to get it back, but it could be a Deis specific thing that no longer works. Good catch.

Reply:
Hmm, only code (in github search of mozilla/mozmeao orgs) reference I found for that header was in basket: https://github.com/mozmeao/basket/blob/81188f37db3deba11243e91534aa9376d36cb7a9/basket/news/views.py#L119

It seems to me (from a really quick look around) this header is valid but maybe old?  A few of the stackoverflow threads seemed to imply you just go down a list of headers till you get the first one. (This one in particular, https://stackoverflow.com/questions/1634782/what-is-the-most-accurate-way-to-retrieve-a-users-correct-ip-address-in-php).

Reply:
It's totally going to depend on our setup. I'll have been forwarded twice for us, once from Cloudflare and then again from the ELB. So we may have to just inspect some requests to see what the real deal is.

Reply:
based on example requests in Sentry it appears that Django does get the `x-forwarded-for` header, and that header can contain comma separated IPs, the first of which is the client IP. I've updated things so that this is supported and tested.